### PR TITLE
HRINT-2417_nodejs_java

### PR DESCRIPTION
**Pull request by Drorlev**
Java and node.js - projecting-using-functions

### CODE REVIEW

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import net.ravendb.demo.common.DocumentStoreHolder;
+
+
+
+import java.util.List;
+
+public class projectingUsingFunctions {
+
+    //region Demo
+    public static class Employee {
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:14]: In the C# demo, this class is called `EmployeeDetails`

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                    "return { Title : formatTitle(employee), FullName : formatName(employee) };" +
+                    "}"+
+        //endregion
+        //region Step_2
+                    "from Employees as employee select output(employee)");
+        //endregion
+        
+        //region Step_3
+            IRawDocumentQuery<Employee> projectedQueryWithFunctions = session.advanced()
+            .rawQuery(Employee.class,rawQueryString);
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:38]: 1. missing space after the comma
2. indent line 38

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        //endregion
+        
+        //region Step_4
+             projectedResults = projectedQueryWithFunctions.toList();
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:42]: take one space to the left

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    public List<Employee> run() {
+        List<Employee> projectedResults;
+        //region Demo
+        try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
+        //region Step_1
+            String rawQueryString =  "declare function output(employee) {" +
+                    "var formatName  = function(employee) { return 'FullName: ' + employee.FirstName + ' ' + employee.LastName; };" +
+                    "var formatTitle = function(employee) { return 'Title: ' + employee.Title };" +
+                    "return { Title : formatTitle(employee), FullName : formatName(employee) };" +
+                    "}"+
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:30]: indentation ?

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        }
+        */
+    ],
+
+    "Walkthroughs": [
+        {
+            "Title": "Define the projecting function",
+            "Slug": "step-1",
+            "DescriptionLines": [
+                "* Within a raw-query-string, define a _JavaScript function_ that processes the 'Employee' document and returns a new 'projected' object.",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json:38]: Add:
...and returns a new 'projected' object in the shape of the 'EmplyeeDetails' class.

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+                    "Type": "link",
+                    "Url": "https://ravendb.net/docs/article-page/latest/java/client-api/session/querying/how-to-project-query-results#example-iv---projection-with-"
+                }
+            ]
+        },
+        {
+            "Title": "Execute the query",
+            "Slug": "step-4",
+            "DescriptionLines": [
+                "* The 'projectedResults' variable only stores the query definition.<br>",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json:84]: change:
'projectedResults' => 'projectedQueryWithFunctions'

pls do this for all 3: C#, java & Go

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    "DescriptionLines": [
+        "The projected query fields can be further processed by using **JavaScript functions.**",
+        "",
+        "These functions are executed on the server side.<br>",
+        "Be aware their execution time does Not time out the query."
+    ],
+
+    "Assets": [
+        {
+            "Title": "Projection with declared function - query on collection",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json:14]: rename title to be correct:
=>
Projection with declared function

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+            "DescriptionLines": [
+                "* Within a raw-query-string, define a _JavaScript function_ that processes the 'Employee' document and returns a new 'projected' object.",
+                "",
+                "* The above _output_ function returns a projected object literal that is computed from the following functions:",
+                "  * formatName -  process the 'FirstName' & 'LastName' document fields.<br>",
+                "  * formatTitle - process the document 'Title' field."
+            ],
+            "Assets": [
+                {
+                    "Title": "Projection with declared function - query on collection",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json:46]: fix the title

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+            "Slug": "step-2",
+            "DescriptionLines": [
+                "* Within the raw-query-string, define the `RQL query` that will be executed on the server.",
+                "",
+                "* Use `select` and call the JavaScript function defined from the previous step.",
+                "",
+                "* In the above RQL, a query is defined on the 'Employees' collection.",
+                "  Note the _select_ usage so that returned results are structured according to the defined _output_ method."
+            ],
+            "Assets": [
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json:63]: you can link to the relevant rql article in docs
https://ravendb.net/docs/article-page/5.3/java/indexes/querying/what-is-rql

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+        },
+        {
+            "Title": "Set the query",
+            "Slug": "step-3",
+            "DescriptionLines": [
+                "* Pass the raw-query-string to the Session `rawQuery()` method."
+            ],
+            "Assets": [
+                {
+                    "Title": "Projection with declared function - query on collection",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json:74]: fix title

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+        {
+            "Title": "Projection with declared function - query on collection",
+            "Type": "link",
+            "Url": "https://ravendb.net/docs/article-page/latest/java/client-api/session/querying/how-to-project-query-results#example-iv---projection-with-"
+        },
+        {
+            "Title": "Projecting With Object Literals",
+            "Type": "link",
+            "Url": "https://ravendb.net/learn/inside-ravendb-book/reader/4.0/9-querying-in-ravendb#projecting-with-object-literals"
+        }
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json:22]: suggesting to add a link to the 'java' Query Overview Demo
(and same for the nodejs)

File: DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+    "DescriptionLines": [
+        "The projected query fields can be further processed by using **JavaScript functions.**",
+        "",
+        "These functions are executed on the server side.<br>",
+        "Be aware their execution time does Not time out the query."
+    ],
+
+    "Assets": [
+        {
+            "Title": "Projection with declared function - query on collection",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json:14]: fix title

File: DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+            "DescriptionLines": [
+                "* Within a raw-query-string, define a _JavaScript function_ that processes the 'Employee' document and returns a new 'projected' object.",
+                "",
+                "* The above _output_ function returns a projected object literal that is computed from the following functions:",
+                "  * formatName -  process the 'FirstName' & 'LastName' document fields.<br>",
+                "  * formatTitle - process the document 'Title' field."
+            ],
+            "Assets": [
+                {
+                    "Title": "Projection with declared function - query on collection",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json:46]: fix title

File: DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+            "DescriptionLines": [
+                "* Within the raw-query-string, define the `RQL query` that will be executed on the server.",
+                "",
+                "* Use `select` and call the JavaScript function defined from the previous step.",
+                "",
+                "* In the above RQL, a query is defined on the 'Employees' collection.",
+                "  Note the _select_ usage so that returned results are structured according to the defined _output_ method."
+            ],
+            "Assets": [
+
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json:65]: link to:
https://ravendb.net/docs/article-page/5.3/nodejs/indexes/querying/what-is-rql

File: DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+        },
+        {
+            "Title": "Set the query",
+            "Slug": "step-3",
+            "DescriptionLines": [
+                "* Pass the raw-query-string to the Session `rawQuery()` method."
+            ],
+            "Assets": [
+                {
+                    "Title": "Projection with declared function - query on collection",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json:76]: fix

**Review by Danielle9897 — CHANGES_REQUESTED** [review]

**Comment by ml054** [comment]
@Drorlev I will review once you finish @Danielle9897 remarks

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java (hunk 1 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:14]: Fixed

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java (hunk 2 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:38]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java (hunk 3 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:42]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java (hunk 4 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:30]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json (hunk 5 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json:38]: Done

File: DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json (hunk 12 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json:14]: Done

File: DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json (hunk 13 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json:46]: Done

File: DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json (hunk 14 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json:65]: Done

File: DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json (hunk 15 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json:76]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java [hunk 16]
```diff
@@ -0,0 +1,48 @@
+//region Usings
+import net.ravendb.client.documents.session.IDocumentSession;
+import net.ravendb.client.documents.session.IRawDocumentQuery;
+//endregion
+import net.ravendb.demo.common.DocumentStoreHolder;
+
+
+
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:8]: Remove the extra empty lines

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+public class projectingUsingFunctions {
+
+    //region Demo
+    public static class EmployeeDetails {
+        public String FullName;
+        public String Title;
+    }
+    //endregion
+
+
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:20]: Remove the extra empty line

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+    public List<EmployeeDetails> run() {
+        List<EmployeeDetails> projectedResults;
+        //region Demo
+        try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
+        //region Step_1
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:25]: indent the `//region Step-x` according to code - see how in other java demos

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+            String rawQueryString =  "declare function output(employee) {" +
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:26]: Remove the extra space after =

Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:26]: format like in the GO demo, without `"` and make similar indentations

image

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json (hunk 6 above)
Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json:104]: pls apply comment

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json (hunk 7 above)
Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json]: pls always write '`done`' if you applied the comment

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java (hunk 17 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:20]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java (hunk 18 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:25]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java (hunk 19 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:26]: Done

**Comment by Danielle9897** [comment]
@Drorlev Pls rebase branch first

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java [hunk 20]
```diff
@@ -0,0 +1,45 @@
+//region Usings
+import net.ravendb.client.documents.session.IDocumentSession;
+import net.ravendb.client.documents.session.IRawDocumentQuery;
+//endregion
+import net.ravendb.demo.common.DocumentStoreHolder;
+import java.util.List;
+
+public class projectingUsingFunctions {
```
Comment (ml054) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:8]: uppercase for java classes (both class and file name)

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+            IRawDocumentQuery<EmployeeDetails> projectedQueryWithFunctions = session.advanced()
+                .rawQuery(EmployeeDetails.class, rawQueryString);
+            //endregion
+        
+            //region Step_4
+            projectedResults = projectedQueryWithFunctions.toList();
+            //endregion
+        //endregion
+    }
+        return projectedResults;
```
Comment (ml054) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:43]: wrong indent?

File: DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/index.js [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+            var formatName  = function(employee) { return "FullName: " + employee.FirstName + " " + employee.LastName; };
+            return { Title : formatTitle(employee), FullName : formatName(employee) };
+        }
+    //endregion
+    //region Step_2
+        from Employees as employee select output(employee)`;
+    //endregion
+
+    //region Step_3
+    const projectedQueryWithFunctions  = await session.advanced.rawQuery(rawQueryString);
```
Comment (ml054) [inline DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/index.js:20]: I don't think we can await on this. (we should await on `all` call below)

Comment (ml054) [reply DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/index.js:20]: also please remove double spaces around `=`

File: DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json [hunk 23]
```diff
@@ -0,0 +1,120 @@
...
+            "DescriptionLines": [
+                "* The 'projectedQueryWithFunctions' variable only stores the query definition.<br>",
+                "  Execute the query by calling `all()`.",
+                "",
+                "* The _projectedResults_ variable will hold the resulting matching entities.",
+                "",
+                "* The output projected entities returned by the query are Not tracked by the Session,<br>",
+                "  as they are not full documents.",
+                "",
+                "* Any change made to these entities will Not be sent to the server upon calling SaveChanges()."
```
Comment (ml054) [inline DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json:114]: SaveChanges() -> saveChanges() - this is node.js

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java (hunk 21 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:43]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java (hunk 20 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:8]: Done

Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java]: Locally .java has changed but git didn't got the change

File: DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/index.js (hunk 22 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/index.js:20]: Done

File: DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json (hunk 23 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/nodejs/demo/queries/projectingUsingFunctions/metadata.json:114]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json [hunk 24]
```diff
@@ -0,0 +1,118 @@
...
+    "Assets": [
+        {
+            "Title": "Query Overview Demo",
+            "Type": "demo",
+            "Url": "/java/queries/query-overview"
+        },
+        {
+            "Title": "What is RQL",
+            "Type": "link",
+            "Url": "https://ravendb.net/docs/article-page/5.3/java/indexes/querying/what-is-rql"
```
[the remark this answers is not in the document]
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json:21]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json [hunk 25]
```diff
@@ -0,0 +1,118 @@
...
+                "* Use `select` and call the JavaScript function defined from the previous step.",
+                "",
+                "* In the above RQL, a query is defined on the 'Employees' collection.",
+                "  Note the _select_ usage so that returned results are structured according to the defined _output_ method."
+            ],
+            "Assets": [
+                {
+                    "Title": "What is RQL",
+                    "Type": "link",
+                    "Url": "https://ravendb.net/docs/article-page/5.3/java/indexes/querying/what-is-rql"
```
[the remark this answers is not in the document]
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/metadata.json:77]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java [hunk 26]
```diff
@@ -0,0 +1,46 @@
...
+    }
+    //endregion
+
+    public List<EmployeeDetails> run() {
+        List<EmployeeDetails> projectedResults;
+        //region Demo
+        try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
+            //region Step_1
+            String rawQueryString = 
+                "declare function output(employee) {\n" +
```
[the remark this answers is not in the document]
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/projectingUsingFunctions/projectingUsingFunctions.java:23]: Done

File: DemoServer/AdditionalLanguages/go/Queries/projectingUsingFunctions/ProjectingUsingFunctions.go [hunk 27]
```diff
@@ -50,7 +50,7 @@ func ProjectingUsingFunctions() error {
     rawQueryString := `
         //region Step_1
         declare function output(employee) {
-            var formatName  = function(employee) { return "Full Name: " + employee.FirstName + " " + employee.LastName; };
```
[the remark this answers is not in the document]
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/go/Queries/projectingUsingFunctions/ProjectingUsingFunctions.go:53]: Done

**Review by Danielle9897 — APPROVED** [review]

**Review by ml054 — APPROVED** [review]

**Pull request merged**